src/PhpPact/Consumer/Matcher/Formatters/PluginFormatter.php

```diff
@@ -40,13 +44,22 @@ public function format(MatcherInterface $matcher): string
         if ($matcher instanceof NullValue) {
             return $this->formatMatchersWithoutConfig(new Type(null));
         }
+        if ($matcher instanceof MinType) {
+            return $this->formatMinTypeMatcher($matcher);
+        }
+        if ($matcher instanceof MaxType) {
+            return $this->formatMaxTypeMatcher($matcher);
+        }
 
         if (in_array($matcher->getType(), self::MATCHERS_WITHOUT_CONFIG)) {
             return $this->formatMatchersWithoutConfig($matcher);
         }
         if ($matcher instanceof AbstractDateTime || $matcher instanceof Regex || $matcher instanceof ContentType) {
             return $this->formatMatchersWithConfig($matcher);
         }
+        if ($matcher instanceof MatchAll) {
+            return $this->formatMatchAllMatchers($matcher);
+        }
```

Comment:
Too much if conditions. I tried to solve this code smell by applying `Visitor` design pattern, but end up reverting the PRs, because there is a problem I can't solve it right now. I guess we will need to solve this one later.

Reply:
Refactored in https://github.com/pact-foundation/pact-php/pull/599.

Instead of applying `Visitor`, I use `Factory Method` (combined with `Bridge`) patterns.

This time, I implemented full code before sending PR to prevent reverting the PRs that are not working.

That's a very big PR. I am not patient so I send it to review without breaking down to smaller PRs. I think it will take 1,5 weeks to review all of them. I'm sorry but I don't have time right now.